docs/typescript/clients.md

```diff
@@ -246,3 +259,97 @@ export async function CancelExternalWorkflow(wfId: string): void {
 ```
 
 Again, see the [Workflow APIs documentation](/docs/typescript/workflows#external-workflows) for full details.
+
+## Advanced: Making raw gRPC calls
+
+Under the hood of a `WorkflowClient`, the `Connection` is actually powered by a `WorkflowService` driver that makes the raw gRPC calls to Temporal Server.
+This Service is capable of making a wider range of introspection calls (as per [the API reference](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#methods)):
+
+```ts
+import { Connection, WorkflowClient } from '@temporalio/client';
+
+const connection = new Connection(); // to configure connection for production
+
+// normal Workflow client
+const client = new WorkflowClient(connection.service);
+await client.start(/* etc */);
+
+// equivalent in raw gRPC calls
+// promise-based API
+await connection.service.startWorkflowExecution(/* etc */);
+// alternate Node callback-style API
+connection.service.startWorkflowExecution(/* options */, callbackFn);
+```
+
+Using gRPC calls is often the only way to access some of the more advanced queries you can make from Temporal Server.
+We highlight some queries of interest here:
+
+<details>
+<summary><a class="font-mono" href="https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#getworkflowexecutionhistory">getWorkflowExecutionHistory</a>
+</summary>
+
+```ts
+const res = await connection.service.getWorkflowExecutionHistory({
+  execution: { workflowId: 'my-business-id' },
+  namespace: 'default',
+});
+console.log(res.history);
+```
+
+Outputs:
+
+```
+{
+  events: [
+    HistoryEvent {
+      eventId: [Long],
+      eventTime: [Timestamp],
+      eventType: 1,
+      taskId: [Long],
+      workflowExecutionStartedEventAttributes: [WorkflowExecutionStartedEventAttributes]
+    },
+    HistoryEvent {
+      eventId: [Long],
+      eventTime: [Timestamp],
+      eventType: 5,
+      taskId: [Long],
+      workflowTaskScheduledEventAttributes: [WorkflowTaskScheduledEventAttributes]
+    },
+    HistoryEvent {
+      eventId: [Long],
+      eventTime: [Timestamp],
+      eventType: 6,
+      taskId: [Long],
+      workflowTaskStartedEventAttributes: [WorkflowTaskStartedEventAttributes]
+    }
+  ]
+}
+```
+
+</details>
+
+<details>
+<summary><a class="font-mono" href="https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#listworkflowexecutions) (requires ElasticSearch">listWorkflowExecutions</a>
```

Comment:
I don't think this requires elasticsearch if you use basic queries.

Reply:
i tried it, it does

Reply:
```suggestion
<summary><a class="font-mono" href="https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#listworkflowexecutions) (requires ElasticSearch)">listWorkflowExecutions</a>
```